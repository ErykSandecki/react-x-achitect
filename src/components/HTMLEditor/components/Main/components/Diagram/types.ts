// types
import { BlockType, PortType, T2DCoordinates, TBlock, TConnector } from 'types';

export type TDraggableConnectorSource = {
  blockId: string;
  blockType: BlockType;
  portType: PortType;
};

export type TDraggableConnectorTarget = {
  blockId: string;
  blockType: BlockType;
  portType: PortType;
};

export type TDraggableConnector = {
  showIndicator: boolean;
  source?: TDraggableConnectorSource;
  sourceCoordinates: T2DCoordinates;
  target?: TDraggableConnectorTarget;
  targetCoordinates: T2DCoordinates;
};

export type TPossiblePort = T2DCoordinates & {
  blockId: string;
  blockType: BlockType;
  height: number;
  portType: PortType;
  width: number;
};

export type TShadowConnector = Pick<
  TConnector,
  'joinOperator' | 'relatedWithBlockId' | 'towardsTo'
> & {
  sourceBlock: Partial<TBlock> & { portType: PortType };
  targetBlock: Partial<TBlock> & { portType: PortType };
};
