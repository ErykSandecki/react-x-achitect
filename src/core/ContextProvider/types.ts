// types
import { Theme } from 'types/enums';

export type TContext = {
  scrollLock: boolean;
  setScrollLock: (scrollLock: boolean) => void;
  setTheme: (theme: Theme) => void;
  theme: Theme;
};

export type TActionOnChangeTheme = (payload: Theme) => {
  payload;
  type: string;
};
