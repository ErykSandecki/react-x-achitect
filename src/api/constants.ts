// types
import { Header } from './enums';

export const BASE_URL = '';

export const HEADERS = {
  [Header.contentType]: 'application/json',
};

export const RESPONSE_DEV_TIMEOUT = 'dev-response-timeout';

export const translationNameSpace = 'response';
