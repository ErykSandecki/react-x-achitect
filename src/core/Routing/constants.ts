export const translationNameSpace = 'routing';
export const guardNameSpace = `${translationNameSpace}.guard`;
