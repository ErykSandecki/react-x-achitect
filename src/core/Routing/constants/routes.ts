export enum RouteName {
  home = 'home',
}

export const ROUTES = {
  [RouteName.home]: '/',
};
