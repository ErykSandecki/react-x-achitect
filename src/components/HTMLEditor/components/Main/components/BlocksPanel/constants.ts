// others
import { translationNameSpace as parentNameSpace } from '../../../../constants';

export const BLOCKS_PANEL_ID = 'blocksPanel';
export const MIN_HEIGHT = 283;

export const translationNameSpace = `${parentNameSpace}.blocksPanel`;
