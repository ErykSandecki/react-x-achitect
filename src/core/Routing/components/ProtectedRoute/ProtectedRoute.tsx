import { FunctionComponent, useEffect } from 'react';
import { Route } from 'react-router-dom';
import { useDispatch } from 'react-redux';

// others
import { guardNameSpace } from '../../constants';

// store
// import { notificationStore } from 'store';

// types
// import { NotificationType } from 'components';
import { TProtectedRouteProps } from '../../types';

const ProtectedRoute: FunctionComponent<TProtectedRouteProps> = ({
  guards = [],
  children,
  ...props
}) => {
  // const { addNotification } = notificationStore.actions;
  const dispatch = useDispatch();
  const failedGuard = guards.find(({ guardCheck }) => !guardCheck());
  const wrapperProps = {
    ...props,
    path: `/${props.path}`,
  };

  useEffect(() => {
    if (failedGuard) {
      const { translationKey } = failedGuard;

      // if (translationKey) {
      //   dispatch(
      //     addNotification({
      //       messageTranslationKey: `${guardNameSpace}.${translationKey}.description`,
      //       notificationType: NotificationType.warning,
      //       titleTranslationKey: `${guardNameSpace}.${translationKey}.title`,
      //     }),
      //   );
      // }
    }
  }, [failedGuard]);

  if (failedGuard) {
    return failedGuard.renderFallback(props);
  }

  return <Route {...wrapperProps}>{children}</Route>;
};

export default ProtectedRoute;
