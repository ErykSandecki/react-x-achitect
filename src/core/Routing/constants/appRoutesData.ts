// components
import HomePage from 'pages/HomePage/HomePage';

// types
import { RouteName } from './routes';
import { TAppRoutesData } from '../types';

export const APP_ROUTES_DATA: TAppRoutesData = [
  {
    Component: HomePage,
    guards: [],
    name: RouteName.home,
  },
];
