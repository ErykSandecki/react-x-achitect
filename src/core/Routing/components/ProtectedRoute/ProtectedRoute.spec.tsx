import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { Redirect, Router } from 'react-router';
import { render } from '@testing-library/react';

// components
import ProtectedRoute from './ProtectedRoute';

// mocks
import { routerStateMock } from 'test/mocks/reducer/routerMock';

// others
import { TAppRouteData } from '../../types';

// store
import { configureStore } from 'store/store';

// types
import { RouteName } from '../../constants/routes';

const history = createMemoryHistory({
  initialEntries: ['*'],
});

const stateMock = {
  ...routerStateMock,
};

const routeData: TAppRouteData = {
  Component: () => <div>Component</div>,
  guards: [
    {
      guardCheck: () => true,
      renderFallback: () => <Redirect exact from={'/'} to={'/error'} />,
      translationKey: '',
    },
  ],
  name: RouteName.home,
};

describe('ProtectedRoute snapshots', () => {
  // mock
  const store = configureStore(stateMock);

  it('should allow user on route', () => {
    // mock
    const { Component, guards, name } = routeData;

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <ProtectedRoute component={Component} guards={guards} name={name} />
        </Router>
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should allow user without guards', () => {
    // mock
    const { Component, name } = routeData;

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <ProtectedRoute component={Component} name={name} />
        </Router>
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should not allow user on route', () => {
    // mock
    const store = configureStore(stateMock);
    const { Component, guards, name } = routeData;

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <ProtectedRoute
            component={Component}
            guards={[{ ...guards![0], guardCheck: () => false }]}
            name={name}
          />
        </Router>
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should not allow user on route with translation key', () => {
    // mock
    const store = configureStore(stateMock);
    const { Component, guards, name } = routeData;

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <ProtectedRoute
            component={Component}
            guards={[
              {
                ...guards![0],
                guardCheck: () => false,
                translationKey: 'translationKey',
              },
            ]}
            name={name}
          />
        </Router>
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
