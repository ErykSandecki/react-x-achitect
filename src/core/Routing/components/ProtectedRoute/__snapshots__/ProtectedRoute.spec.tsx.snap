// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedRoute snapshots should allow user on route 1`] = `<DocumentFragment />`;

exports[`ProtectedRoute snapshots should allow user without guards 1`] = `<DocumentFragment />`;

exports[`ProtectedRoute snapshots should not allow user on route 1`] = `<DocumentFragment />`;

exports[`ProtectedRoute snapshots should not allow user on route with translation key 1`] = `<DocumentFragment />`;
